lib/auth/session_access.go

```diff
@@ -209,6 +221,39 @@ func (e *SessionAccessEvaluator) hasPolicies() bool {
 	return false
 }
 
+// Generate a pretty-printed string of precise requirements for session start suitable for user display.
+func (e *SessionAccessEvaluator) PrettyRequirementsList() string {
+	s := "require all:"
+
+	for _, policySet := range e.policySets {
+		policies := e.extractApplicablePolicies(policySet)
+		if len(policies) == 0 {
+			continue
+		}
+
+		s += fmt.Sprintf("\n\t  one of (%v):", policySet.Name)
```

Comment:
This is okay, but [`strings.Builder`](https://pkg.go.dev/strings#Builder) would be better than a bunch of Sprintf and concatenation.